plugins/org.eclipse.n4js/src/org/eclipse/n4js/validation/validators/packagejson/N4JSProjectSetupJsonValidatorExtension.xtend

```diff
@@ -359,30 +364,39 @@ public class N4JSProjectSetupJsonValidatorExtension extends AbstractPackageJSONV
 		return types;
 	}
 	
-	/** IDEBUG-266 issue error warning on cyclic dependencies. */
-	// TODO: GH-1500: Check if cyclic dependencies are already found in BuildManager#sortByDependencies()
-	// @Check // Commented out because there are performance issues.
-	def checkCyclicDependencies(JSONDocument document) {
-		val project = findProject(document.eResource.URI).orNull;
-		if (null !== project) {
-			
-			val dependencyProvider = new SourceContainerAwareDependencyProvider(true);
-			val traverser = new DependencyTraverser(project, dependencyProvider, true);
-			
-			val traversalResult = traverser.findCycle();
-			
-			if (traversalResult.hasCycle) {
-				// add issue to 'name' property or alternatively to the whole document
-				val nameValue = getSingleDocumentValue(NAME);
-				val message = getMessageForPROJECT_DEPENDENCY_CYCLE(traversalResult.prettyPrint([calculateName.rawName]));
-				addIssuePreferred(#[nameValue], message, PROJECT_DEPENDENCY_CYCLE);
-			} else {
-				//for performance reasons following is not separate check
-				/*
-				 * otherwise we would traverse all transitive dependencies multiple times,
-				 * and we would care for cycles
-				 */
-				project.holdsProjectWithTestFragmentDependsOnTestLibrary()
+	@CheckProperty(property = DEPENDENCIES)
+	def checkCyclicDependencies(JSONValue dependenciesValue) {
+		// exit early in case of a malformed dependencies section (structural validation is handled elsewhere)
+		if (!(dependenciesValue instanceof JSONObject)) {
+			return;
+		}
+		
+		// pairs that represent project dependencies
+		val dependencyPairs = (dependenciesValue as JSONObject).nameValuePairs;
+		// obtain project description for higher-level access to contained information
+		val projectName = getProjectDescription().projectName;
+		// get project build order of current build
+		val projectOrderInfo = workspaceConfigProvider.get.getProjectBuildOrderInfo;
```

Comment:
Accessing this global value here is not ideal. For example, in editors a change of the workspace configuration will show up at arbitrary points in time (not just when the corresponding event is received).

We should make the workspace config snapshot available via the resource set and access it from there. Not part of this user story, though.